examples/fx_gan.py

```diff
@@ -0,0 +1,144 @@
+
+feat_length = 168
+input_size = 120
+feat_channel = 10
+epochs = 100
+
+class out_layer(nn.Module):
+    def __init__(self, in_channels, out_channels = 1, gap_size = feat_length):
+        super(out_layer, self).__init__()
+        self.in_channels = in_channels
+        self.gap = nn.AvgPool1d(kernel_size = gap_size, padding = 1)
+        self.lin = nn.Linear(in_channels, out_channels)
+        self.activation = nn.Sigmoid()
+    def forward(self, x):
+        gap = self.gap(x).view(-1, self.in_channels)
+        return self.activation(self.lin(gap))
+
+class base_conv1d(nn.Module):
+    def __init__(self, in_channels, out_channels, ks = 3, pd = 1, do = 0.0, **kwargs):
+        super(base_conv1d, self).__init__()
+        self.conv = nn.Conv1d(in_channels, out_channels, 
+            bias = False, kernel_size = ks, padding=pd, **kwargs)
+        self.activation = nn.RReLU(inplace = True)
+        self.conv.weight.data.normal_(0, 0.01)
+    def forward(self, x):
+        return self.activation(self.conv(x))
+
+# Discriminator needs to take the generator output or the real sample and guess whether fake or not
+class Discriminator(nn.Module):
+    def __init__(self, channels = 1, out_channels = 1, gap_size = feat_length, noise = 0.1):
+        super(Discriminator, self).__init__()
+        self.noise = noise
+        self.conv1 = base_conv1d(in_channels = channels, out_channels = 512, ks = 7, pd = 3)
+        self.conv2 = base_conv1d(in_channels = 512, out_channels = 256, ks = 5, pd = 2)
+        self.conv3 = base_conv1d(in_channels = 256, out_channels = 128)
+        self.out = out_layer(in_channels = 128, out_channels = out_channels, gap_size = gap_size)
+    def forward(self, x):
+        if self.training and self.noise != 0.0:
+            x = x + Variable(x.data.new(x.size()).normal_(0.0, self.noise))
+        return self.out(self.conv3(self.conv2(self.conv1(x))))
+
+# Generator needs to take the latent variables and then generate a fake sequence of given channels
+class Generator(nn.Module):
+    def __init__(self, channels = 1, gap_size = feat_length):
+        super(Generator, self).__init__()
+        self.channels = channels
+        self.gap_size = gap_size
+        self.conv1 = base_conv1d(in_channels = 128, out_channels = 256)
+        self.conv2 = base_conv1d(in_channels = 256, out_channels = 512, ks = 5, pd = 2)
+        self.conv3 = base_conv1d(in_channels = 512, out_channels = feat_length-input_size, ks = 7, pd = 3)
+        self.conv3.activation = nn.Sigmoid() # Because the input is in [0, 1]
+    def forward(self, x, direction = True):
+        gen = Variable(torch.randn(x.shape[0], 128, self.channels)).cuda()
+        gen = self.conv3(self.conv2(self.conv1(gen))).view(-1, self.channels, feat_length-input_size)
+        return torch.cat([gen, x], dim=2) if direction else torch.cat([x, gen], dim=2)
+
+Dataset = pd.read_csv('Dataset.csv', index_col='Gmt time', parse_dates=True, infer_datetime_format=True)
```

Comment:
If we are going to have this in the examples the dataset should be downloadable from somewhere and we should have proper license to do so. If this one doesn't conform to these, we have to use some other dataset.